Sonarqube 'auto-discover' all components (#13450)
DataDog/integrations-core: sonarqube/README.md

<!-- xxz tab xxx -->
<!-- xxz tabs xxx -->

#### Components Discovery

You can configure how your components are discovered with the `components_discovery` parameter.

> `limit`
> : Maximum number of items to be auto discovered.  
> **Default value**: `10`
> 
> `include`
> : Mapping of regular expressions keys and component config values that will be auto discovered.  
> **Default value**: empty map
> 
> `exclude`
> : List of regular expressions with the patterns of components that will not be auto discovered.  
> **Default value**: empty list

Reviewer: ```suggestion
`limit`
: Maximum number of items to be autodiscovered.  
**Default value**: `10`

`include`
: Mapping of regular expression keys and component config values to autodiscover.  
**Default value**: empty map

`exclude`
: List of regular expressions with the patterns of components to exclude from autodiscovery.  
**Default value**: empty list
```

> Examples:

Reviewer: ```suggestion
**Examples**:
```

Include a maximum of `5` components with names starting with `my_project`:

```yaml
components_discovery:
  limit: 5
  include:
    'my_project*':
```

Include a maximum of `20` components and exclude those beginning with `temp`:

```yaml
components_discovery:
  limit: 20
  include:

>     '*':

Reviewer 2: This is invalid. All `*` should be `.*`

  exclude:
    - 'temp*'
```

> Include all components with names starting with `issues` processing them with `issues_project`, and only collect metrics belonging to the category `issues`. As `limit` is not defined, the number of components discovered is limited to the default value `10`:

Reviewer 2: `issues` apply -> `issues`, apply ?

> Include all components with names starting with `issues` processing them with `issues_project`, and only collect metrics belonging to the category `issues`. As `limit` is not defined, the number of components discovered is limited to the default value `10`:

Reviewer: Apologies if my [initial suggestion](https://github.com/DataDog/integrations-core/pull/13450#discussion_r1039832655) would have introduced inaccuracy here -- is the following text meant to indicate that only components with the `issues_project` tag are included?
> processing them with `issues_project`

Or does using the `tag` key do something else, such as apply the tag?

Author: Yes, the `tag` key indicates which tag will be applied

```yaml
components_discovery:
  include:
    'issues*':
       tag: issues_project
       include:

>          - issues.

Reviewer 2: ```suggestion
         - issues\.
```

Reviewer 2: https://github.com/DataDog/integrations-core/pull/13450#pullrequestreview-1208727224

See [service_checks.json][15] for a list of service checks provided by this integration.

### Components Discovery

> The user can configure the discovery functionality for components: `components_discovery`

Reviewer: ```suggestion
You can configure how your components are discovered with the `components_discovery` parameter.
```

> `limit`: Maximum number of items to be auto discovered. Default value: `10`
> 
> `include`: Mapping of regular expressions keys and component config values that will be auto discovered. Default value: empty map
> 
> `exclude`: List of regular expressions with the patterns of components that will not be auto discovered. Default value: empty list

Reviewer: ```suggestion
`limit`
: Maximum number of items to be auto discovered.  
**Default value**: `10`

`include`
: Mapping of regular expressions keys and component config values that will be auto discovered.  
**Default value**: empty map

`exclude`
: List of regular expressions with the patterns of components that will not be auto discovered.  
**Default value**: empty list
```

Examples:

> Include only `components` whose name starts with `my_project` up to a maximum of `5`:

Reviewer: ```suggestion
Include a maximum of `5` components with names starting with `my_project`:
```

```yaml
components_discovery:
  limit: 5
  include:

>     'my_project*':

Reviewer 3: What does this map to?
